composer/spec/dependabot/composer/file_updater/lockfile_updater_spec.rb

```diff
@@ -273,23 +289,54 @@
         it "runs just fine (we get a 404 here because our key is wrong)" do
           expect { updated_lockfile_content }.to raise_error do |error|
             expect(error).to be_a(Dependabot::DependencyFileNotResolvable)
-            expect(error.message).to include("404 Not Found")
+            expect(error.message).to include("404")
           end
         end
       end
     end
 
-    context "with a path source" do
-      let(:files) { [composer_json, lockfile, path_dep] }
-      let(:manifest_fixture_name) { "path_source" }
-      let(:lockfile_fixture_name) { "path_source" }
-      let(:path_dep) do
-        Dependabot::DependencyFile.new(
-          name: "components/path_dep/composer.json",
-          content: fixture("composer_files", "path_dep")
-        )
+    context "that requires an environment variable (composer v2)" do
+      let(:project_name) { "env_variable" }
+
+      context "that hasn't been provided" do
+        it "raises a DependencyFileNotResolvable error" do
+          expect { updated_lockfile_content }.to raise_error do |error|
+            expect(error).to be_a(Dependabot::DependencyFileNotResolvable)
```

Comment:
TODO: figure out if we can improve this error message for v2, this used to raise (see v1 example)

```
expect(error).to be_a(Dependabot::MissingEnvironmentVariable)
expect(error.environment_variable).to eq("ACF_PRO_KEY")
```

Reply:
Fixed was caused by a plugin that only worked for composer v1